### ISSUE TITLE
My Average loss is not decreasing in yolov4-csp?

### ISSUE THREAD

**Issue by akashAD98** [AlexeyAB/darknet#7674]
I have 29 classes, so my max_batches=(29*2000=58,0000) so I trained my model for 58000 steps  & my avg_loss is 24. So my question is can increase the max_batches & training steps??  Will my loss decrease after few steps ??? the learning rate & max_delta will affect the model loss?? i selected max_delta=5 in last [yolo] layer & my lr_rate is 0.0001
please anyone can clarify my doubt. Thanks in advance.

**Comment by poornimajd** [comment]
Depends on how challenging is your dataset and also number of images you have.

**Comment by PedroBarcha** [comment]
> I have 29 classes, so my max_batches=(29*2000=58,0000) so I trained my model for 58000 steps & my avg_loss is 24. So my question is after increasing the max_batches & training steps?? Will my loss decrease after few steps ??? the learning rate & max_delta will affect the model loss?? i selected max_delta=5 in last [yolo] layer & my lr_rate is 0.0001
> please anyone can clarify my doubt. Thanks in advance.

Same here, only when training with CSP. Found out why?
x-special/nautilus-clipboard
copy
![index](https://user-images.githubusercontent.com/11379168/130045250-06a2f57a-01a7-429d-8c95-55bbcba0e941.png)

**Comment by vokhidovhusan** [comment]
> I have 29 classes, so my max_batches=(29*2000=58,0000) so I trained my model for 58000 steps & my avg_loss is 24. So my question is after increasing the max_batches & training steps?? Will my loss decrease after few steps ??? the learning rate & max_delta will affect the model loss?? i selected max_delta=5 in last [yolo] layer & my lr_rate is 0.0001 please anyone can clarify my doubt. Thanks in advance.

I have facing the same issue. I avg loss also around 30 after 5k-10k iteration
